langserve/playground/src/components/ChatMessageTuplesControlRenderer.tsx

```diff
@@ -54,57 +66,59 @@ export const ChatMessageTuplesControlRenderer = withJsonFormsControlProps(
           </label>
           <button
             className="p-1 rounded-full"
-            onClick={() => {
-              const lastRole = data.length ? data[data.length - 1][0] : "ai";
-              props.handleChange(props.path, [
-                ...data,
-                [lastRole === "ai" ? "human" : "ai", ""],
-              ]);
-            }}
+            onClick={() => props.handleChange(props.path, [...data, ["", ""]])}
           >
             <PlusIcon className="w-5 h-5" />
           </button>
         </div>
 
         <div className="flex flex-col gap-3 mt-1 empty:hidden">
-          {data.map(([type, content], index) => {
+          {data.map(([human, ai], index) => {
```

Comment:
I think Eugene brought this up but could be other chat representations of tuples?

Reply:
When discussed with Eugene, all of the existing templates using Tuple[str, str] for turnstile conversation at the moment. if we need to work with flat messages instead, I think we should point users to use List[BaseMessage] instead.